### PR TITLE
#19285 Using Max contentlet properties by Persona

### PR DESCRIPTION
**Pull request by freddyucv**
Max Contentlet container property now is working by Persona

### CODE REVIEW

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 1]
```diff
@@ -1653,6 +1658,103 @@ public void pageWithSidebarShouldRender() throws Exception{
...
+        final HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+
+        final String htmlWithDefaultPersona = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                PageContextBuilder.builder()
+                        .setUser(systemUser)
+                        .setPageUri(page.getURI())
+                        .setPageMode(PageMode.LIVE)
+                        .build(),
+                mockRequest, mockResponse);
+        Assert.assertTrue(htmlWithDefaultPersona.contains("dot:persona"));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1742]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java [hunk 2]
```diff
@@ -514,4 +510,40 @@ private String getPersonaTagToIncludeContent(final HttpServletRequest request, f
 
         return hasPersonalizations ? currentPersonaTag : MultiTree.DOT_PERSONALIZATION_DEFAULT;
     }
+
+    private static class ContainerUUIDPersona {
+        Map<String, List<String>> containerUuidPersona     = Maps.newHashMap();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:515]: Codacy found an issue: [It is somewhat confusing to have a field name matching the declaring class name](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 3]
```diff
@@ -1653,6 +1658,103 @@ public void pageWithSidebarShouldRender() throws Exception{
...
+        mockRequest.getSession().setAttribute(WebKeys.VISITOR, visitor);
+
+        final String htmlWithNoDefaultPersona = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                PageContextBuilder.builder()
+                        .setUser(systemUser)
+                        .setPageUri(page.getURI())
+                        .setPageMode(PageMode.LIVE)
+                        .build(),
+                mockRequest, mockResponse);
+        Assert.assertTrue(htmlWithNoDefaultPersona.contains(notDefaultPersona.getKeyTag()));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1755]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java [hunk 4]
```diff
@@ -1653,6 +1658,103 @@ public void pageWithSidebarShouldRender() throws Exception{
         assertTrue(firstIndex != -1 && secondIndex != -1 && firstIndex != secondIndex && thirdIndex == -1);
     }
 
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPI#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
+     * When: A page has multiple personas version, and the container has a MAx contentlet equals to 1
+     * Should: render the page
+     */
+    @Test
+    public void pageWithMultiplePersona() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1667]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java [hunk 5]
```diff
@@ -514,4 +510,40 @@ private String getPersonaTagToIncludeContent(final HttpServletRequest request, f
...
+        }
+
+        public long getSize(
+                final Container container,
+                final String uniqueUUIDForRender,
+                final PersonalizedContentlet personalizedContentlet) {
+            return get(container, uniqueUUIDForRender, personalizedContentlet).size();
+        }
+
+        public Iterable<? extends Map.Entry<String, List<String>>> entrySet() {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:545]: should return a Set?

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java [hunk 6]
```diff
@@ -514,4 +510,40 @@ private String getPersonaTagToIncludeContent(final HttpServletRequest request, f
 
         return hasPersonalizations ? currentPersonaTag : MultiTree.DOT_PERSONALIZATION_DEFAULT;
     }
+
+    private static class ContainerUUIDPersona {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:514]: Add some doc

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java:517]: done https://github.com/dotCMS/core/pull/19447/commits/bbd176303ed633ddea71b2af08cb868208201451#diff-135d6f9eb01eb89ef742b83753636596a41a9e3ab757abd4347b57e80ae892e0R517

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java (hunk 5 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java]: done https://github.com/dotCMS/core/pull/19447/commits/bbd176303ed633ddea71b2af08cb868208201451#diff-135d6f9eb01eb89ef742b83753636596a41a9e3ab757abd4347b57e80ae892e0R548

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1667]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java:1667]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6340183)

**Pull request merged**